### PR TITLE
move s-mode feature into `app.toml`

### PR DESCRIPTION
**Pull request by beezow**
Moving the feature flags out of the `cargo.toml` and into the `app.toml` makes it significantly easier for humility to decode.

I selected @tdewey-rivos supervisor branch as the merge target to represent the merge dependency.

### CODE REVIEW

**Review by FawazTirmizi — DISMISSED** [review]

**Comment by furquan-rivos** [comment]
Change looks good to me!

**Comment by beezow** [comment]
Ill leave it up to @tdewey-rivos if he wants to merge this into his s-mode work directly, or wait till s-mode gets merged into main.

**Review by ghost — DISMISSED** [review]
Excellent - a much better idea.

**Comment by ghost** [comment]
I'm going to hold off on merging this while I integrate the reduced-fix for Nix, and then rebase the whole branch. Thanks again for looking over this!

**Review by ghost — DISMISSED** [review]

**Comment by beezow** [comment]
Corresponding humility support [here](https://github.com/rivosinc/humility/pull/23)

**Comment by furquan-rivos** [comment]
This PR needs to be rebased now that the supervisor change has landed in hubris.

**Comment by beezow** [comment]
> This PR needs to be rebased now that the supervisor change has landed in hubris.

Done. Also had to propagate the feature flag to the test-suite.

**Review by FawazTirmizi — APPROVED** [review]

**Review by furquan-rivos — APPROVED** [review]

**Pull request merged**
